setup.py

```diff
@@ -158,9 +158,14 @@ def build_extension(self, ext: CMakeExtension):
 
         # Ensure that catalyst shared object is copied to the build directory for pip editable install
         if backend in ("lightning_kokkos"):
-            source = os.path.join(f"{extdir}", f"lib{backend}_catalyst.so")
-            destination = os.path.join(os.getcwd(), "build")
-            shutil.copy(source, destination)
+            if platform.system() == "Linux":
+                source = os.path.join(f"{extdir}", f"lib{backend}_catalyst.so")
+                destination = os.path.join(os.getcwd(), "build")
+                shutil.copy(source, destination)
+            if platform.system() == "Darwin":
+                source = os.path.join(f"{extdir}", f"lib{backend}_catalyst.dylib")
+                destination = os.path.join(os.getcwd(), "build")
+                shutil.copy(source, destination)
```

Comment:
```suggestion
            if platform.system() in ["Linux", "Darwin"]:
                shared_lib_ext = {"Linux": ".so", "Darwin": ".dylib"}
                source = os.path.join(f"{extdir}", f"lib{backend}_catalyst{shared_lib_ext}")
                destination = os.path.join(os.getcwd(), "build")
                shutil.copy(source, destination)
```
I'm assuming here that you tested it both for Linux and Mac systems.

Reply:
wait!

Reply:
OK, you can proceed now.

Reply:
Yes it's tested for Linux and Mac, thanks @AmintorDusko!